src/server/util/domain.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Extracts the root domain from an input url.
+ * @param url The url for its root domain to be extracted.
+ * @returns The root domain is url is valid, else the unchanged url.
+ */
+export default function parseDomain(url: string): string {
+  const regex: RegExp = /^((http|https):\/\/)([^/]*)/
```

Comment:
Can we not use the standard `url` library from NodeJS?

Reply:
Sounds good to me, will work on it

Reply:
Converted to using the `url` library in commit c6c9a28. Added fallback for invalid url in commit f375729.